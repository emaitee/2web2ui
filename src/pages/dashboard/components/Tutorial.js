import React, { Component } from 'react';
import { Panel } from '@sparkpost/matchbox';
import TutorialItem from './TutorialItem';

const sendingDomainUrl = 'account/sending-domains';

export class Tutorial extends Component {
  render() {
    const {
      currentUser,
      hasSendingDomains,
      hasVerifiedDomains,
      hasApiKeysForSending,
      hasBounceDomains,
      hasSentThisMonth
    } = this.props;

    return (
      <Panel title='Getting Started Checklist'>
        <Panel.Section>
          <TutorialItem
            label='Verify your email address'
            completed={currentUser.email_verified} >
            <p>Check your email to verify your email address and increase your daily sending limits</p>
          </TutorialItem>
        </Panel.Section>

        <Panel.Section>
          <TutorialItem
            label='Add a sending domain'
            labelLink={sendingDomainUrl}
            completed={hasSendingDomains}>
            <p>Add a sending domain to send email from your own domain</p>
          </TutorialItem>
        </Panel.Section>

        <Panel.Section>
          <TutorialItem
            label='Verify your sending domain'
            labelLink={sendingDomainUrl}
            completed={hasVerifiedDomains}>
            <p>You'll need to verify your domain before you can use it</p>
          </TutorialItem>
        </Panel.Section>

        <Panel.Section>
          <TutorialItem
            label='Create bounce domain'
            labelLink={sendingDomainUrl}
            completed={hasBounceDomains}>
            <p>Create a bounce domain to track email engagement</p>
          </TutorialItem>
        </Panel.Section>

        <Panel.Section>
          <TutorialItem
            label='Create an API key'
            labelLink='account/api-keys'
            completed={hasApiKeysForSending}>
            <p>You'll need to create an API key to use our API or SMTP integration.</p>
          </TutorialItem>
        </Panel.Section>

        <Panel.Section>
          <TutorialItem
            label='Send an email this month'
            completed={hasSentThisMonth}>
            <p>Get sending!</p>
          </TutorialItem>
        </Panel.Section>
      </Panel>
    );
  }
}

export default Tutorial;
