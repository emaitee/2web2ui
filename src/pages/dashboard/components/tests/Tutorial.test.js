import { shallow } from 'enzyme';
import React from 'react';

import Tutorial from '../Tutorial';

describe('Component: Tutorial', () => {
  const props = {
    currentUser: {
      email_verified: true
    },
    hasSendingDomains: false,
    hasVerifiedDomains: true,
    hasApiKeysForSending: false,
    hasBounceDomains: true
  };

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Tutorial {...props} />);
  });

  it('should render correctly by default', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
