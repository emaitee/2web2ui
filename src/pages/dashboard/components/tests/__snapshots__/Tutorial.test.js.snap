// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Tutorial should render correctly by default 1`] = `
<Panel
  title="Getting Started Checklist"
>
  <Panel.Section>
    <TutorialItem
      completed={true}
      label="Verify your email address"
    >
      <p>
        Check your email to verify your email address and increase your daily sending limits
      </p>
    </TutorialItem>
  </Panel.Section>
  <Panel.Section>
    <TutorialItem
      completed={false}
      label="Add a sending domain"
      labelLink="account/sending-domains"
    >
      <p>
        Add a sending domain to send email from your own domain
      </p>
    </TutorialItem>
  </Panel.Section>
  <Panel.Section>
    <TutorialItem
      completed={true}
      label="Verify your sending domain"
      labelLink="account/sending-domains"
    >
      <p>
        You'll need to verify your domain before you can use it
      </p>
    </TutorialItem>
  </Panel.Section>
  <Panel.Section>
    <TutorialItem
      completed={true}
      label="Create bounce domain"
      labelLink="account/sending-domains"
    >
      <p>
        Create a bounce domain to track email engagement
      </p>
    </TutorialItem>
  </Panel.Section>
  <Panel.Section>
    <TutorialItem
      completed={false}
      label="Create an API key"
      labelLink="account/api-keys"
    >
      <p>
        You'll need to create an API key to use our API or SMTP integration.
      </p>
    </TutorialItem>
  </Panel.Section>
  <Panel.Section>
    <TutorialItem
      label="Send an email this month"
    >
      <p>
        Get sending!
      </p>
    </TutorialItem>
  </Panel.Section>
</Panel>
`;
